processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorsBuilder.java

```diff
@@ -116,15 +127,24 @@ public ProcessorsBuilder<T> thenProcess(DecatonProcessor<T> processor) {
         return thenProcess(new DecatonProcessorSupplierImpl<>(() -> processor, ProcessorScope.PROVIDED));
     }
 
-    Processors<T> build(DecatonProcessorSupplier<byte[]> retryProcessorSupplier) {
-        return new Processors<>(suppliers, retryProcessorSupplier, taskExtractor, retryTaskExtractor);
+    Processors<T> build(DecatonProcessorSupplier<byte[]> retryProcessorSupplier, ProcessorProperties properties) {
```

Comment:
I thought that just taking an instance of `taskExtractor` as an argument of this method was ok, better to do it with "constructor" thing?

Reply:
To instantiate TaskExtractor on `build`'s caller (i.e. SubscriptionBuilder), ProcessorsBuilder needs to store `Deserializer` or `TaskExtractor` passed from user through `consuming` and SubscriptionBuilder need to instantiate appropriate resulting TaskExtractor accordingly, which is fairly complicated.

Maybe just deferring the instantiation by constructor is straightforward.